src/averageAges.js

```diff
@@ -55,12 +85,35 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const peopleWithMother = people.filter(({ mother }) =>
+    mother !== null && people.find(person => person.name === mother));
```

Comment:
The `name` field can also be destructuring.

Reply:
done